scripts/gen-s-parser.py

```diff
@@ -355,19 +335,24 @@ def instruction_parser():
   for inst, expr in instructions:
     inst_length = max(inst_length, len(inst))
     trie.insert(inst, expr)
-  trie.prune()
 
   printer = CodePrinter()
 
   printer.print_line("char op[{}] = {{'\\0'}};".format(inst_length + 1))
   printer.print_line("strncpy(op, s[0]->c_str(), {});".format(inst_length))
 
+  def make_leaf(expr, inst):
+    return ("if (strcmp(op, \"{inst}\") == 0) return {expr}; else break;"
```

Comment:
can use strncmp?

Reply:
Not necessary because the `op` buffer is always null terminated. I could definitely change it if it's a matter of principle, though.

Reply:
I think make sense use pointer + offset like:
```c++
switch (op[1]) {
    case 'l': if (strcmp(op + 2, "block") == 0) ...
```

Reply:
@MaxGraey Most instructions wouldn't benefit much from that because the parser skips checking many intermediate characters, so I'm not sure it's worth the added complexity. If we really want to push performance here there are bigger algorithmic wins available to us such as choosing which character to look at based on which will discriminate the most instructions.

Reply:
Could use something like gperf too. :-)

Reply:
Got it! Another suggestion use less nested switch/case approach but more nested if-else + divide & conquer:
```c++
switch (op[0]) {
   case 'b': {
      int cmp = strcmp(op, "block");
      if (cmp == 0) { makeBlock }
      if (cmp > 0) {
         cmp = strcmp(op, "br");
         if (cmp == 0) { makeBreak }
         if (cmp > 0) {
            if (strcmp(op, "br_if") { ... }
         }
      }
   }
}
```
wdyt?

Reply:
@MaxGraey that looks like it would work, but the logic for deciding when to use a switch and when to use an if/else looks like it would be more complicated than the current solution that always uses a switch. I think we can go with the current solution for now, but if you want to play around with it and see if you can get a perf win with a different solution we can definitely swap it in.